pkg/fission-cli/cmd/spec/apply.go

```diff
@@ -185,6 +188,27 @@ func (opts *ApplySubCommand) run(input cli.Input) error {
 	return nil
 }
 
+func warnIfDirtyWorkTree(path string) {
+	repo, err := git.PlainOpen(path)
+	if err != nil {
+		return
+	}
+
+	workTree, err := repo.Worktree()
+	if err != nil {
+		return
+	}
+
+	status, err := workTree.Status()
+	if err != nil {
+		return
+	}
+
+	if !status.IsClean() {
+		console.Warn("Worktree is not clean")
```

Comment:
We are giving a warning to the user, we should also provide an action to resolve them.
```
worktree is not clean, please ensure you have committed changes to git.
```